pkg/table/tables/tables.go

```diff
@@ -1825,7 +1825,11 @@ func (t *TableCommon) calcChecksums(sctx sessionctx.Context, h kv.Handle, data [
 	}
 	checksums := make([]uint32, len(data))
 	for i, cols := range data {
-		row := rowcodec.RowData{Cols: cols, Data: buf}
+		row := rowcodec.RowData{
+			Cols: cols,
+			Data: buf,
+			Loc:  sctx.GetSessionVars().TimeZone,
```

Comment:
To be consistent with `Encode`, using `sctx.GetSessionVars().StmtCtx.TimeZone()` is better.